### PR TITLE
Feature - Update confirm screen

### PR DESCRIPTION
**Pull request by at-haingo**
#### Related issues (Optional)
Task #70

#### Description (Required)
- [x] Update current in confirm screen when move

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 1]
```diff
@@ -31,20 +37,64 @@ import java.util.*
...
+                    .addConnectionCallbacks(this)
+                    .addApi(LocationServices.API)
+                    .build()
+        }
+    }
+
+    private fun initLocationRequest() {
+        mLocationRequest = LocationRequest.create()
+        mLocationRequest?.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
+        mLocationRequest?.interval = 500
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 2]
```diff
@@ -31,20 +37,64 @@ import java.util.*
...
+        mLocationRequest?.fastestInterval = 500
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 3]
```diff
@@ -110,20 +164,20 @@ class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
                 .anchor(0.5f, 0.5f))
                 ?.showInfoWindow()
         imgPickLocation.visibility = View.INVISIBLE
-        mGoogleMap?.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 16f))
+        mGoogleMap?.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 14f))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 4]
```diff
@@ -110,20 +164,20 @@ class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
...
-            val currentLocation = LatLng(location.latitude, location.longitude)
-            mGoogleMap?.addMarker(MarkerOptions()
+            mMarker = mGoogleMap?.addMarker(MarkerOptions()
                     .position(currentLocation)
                     .icon(BitmapDescriptorFactory.fromResource(R.drawable.ic_current_point))
                     .title(getString(R.string.current_location))
                     .anchor(0.5f, 0.5f))
-            mGoogleMap?.moveCamera(CameraUpdateFactory.newLatLngZoom(currentLocation, 16f))
             addPulseRing(currentLocation)
+            mGoogleMap?.moveCamera(CameraUpdateFactory.newLatLngZoom(currentLocation, 16f))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: build.gradle [hunk 5]
```diff
@@ -7,7 +7,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.0-beta7'
+        classpath 'com.android.tools.build:gradle:3.0.0'
```
Comment (at-quocnguyen) [inline build.gradle]: change cái này làm j thế  bỏ lại đi @@@

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 6]
```diff
@@ -31,20 +37,67 @@ import java.util.*
  * Created by haingoq on 10/10/2017.
  */
 class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
-        LocationSource.OnLocationChangedListener,
+        com.google.android.gms.location.LocationListener,
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: `LocationListener,` thế này đc rồi làm j mừ dài thế ??? com.google.android.gms.location.LocationListener,
với cho nó nàm trên lun khi nào vượt qua cái đường viền mới xún hàng nha

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 7]
```diff
@@ -31,20 +37,67 @@ import java.util.*
...
+        GoogleApiClient.ConnectionCallbacks,
         GoogleMap.OnCameraIdleListener, View.OnClickListener {
+    companion object {
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: break line cho dễ nhìn

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 8]
```diff
@@ -31,20 +37,67 @@ import java.util.*
  * Created by haingoq on 10/10/2017.
  */
 class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
-        LocationSource.OnLocationChangedListener,
+        com.google.android.gms.location.LocationListener,
+        GoogleApiClient.ConnectionCallbacks,
         GoogleMap.OnCameraIdleListener, View.OnClickListener {
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: ko cần break line @@

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 9]
```diff
@@ -31,20 +37,67 @@ import java.util.*
...
+    companion object {
+        private const val INTERVAL = 500L
+    }
+
     private var mGoogleMap: GoogleMap? = null
     private var mMapFragment: SupportMapFragment? = null
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: used `lateinit`

Comment (at-quocnguyen) [reply app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: check dum a may biến phía dưới lun dùng lateinit hết

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 10]
```diff
@@ -31,20 +37,67 @@ import java.util.*
...
+    }
+
+    private fun initLocationRequest() {
+        mLocationRequest = LocationRequest.create()
+        mLocationRequest?.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
+        mLocationRequest?.interval = INTERVAL
+        mLocationRequest?.fastestInterval = INTERVAL
+    }
+
+    @SuppressLint("MissingPermission")
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: chỗ này cần check permisson cho nó nhé !
```kotlin
override fun onConnected(p0: Bundle?) {
        if (!HyperTrack.checkLocationPermission(activity)) {
            HyperTrack.requestPermissions(activity)
        }

        if (!HyperTrack.checkLocationServices(activity)) {
            HyperTrack.requestLocationServices(activity)
        }

        if (HyperTrackUtils.isInternetConnected(activity)) {
            if (HyperTrackUtils.isLocationEnabled(activity)) {
                LocationServices.FusedLocationApi
                        .requestLocationUpdates(mGoogleApiClient, mLocationRequest, this)
            }
        }
    }
```

Comment (at-quocnguyen) [reply app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: lần sau đưng tuỳ tiện dùng thằng này nữa nha `@SuppressLint("MissingPermission")` nếu nó báo lỗi thì tìm cách fix còn warning thì để đó push lên mừ tốt nhất tìm cách fix chứ để thằng này zô nghĩa là nó bỏ qua rất nguy hiểm cho app sau này

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 11]
```diff
@@ -67,7 +124,7 @@ class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
     override fun onClick(view: View?) {
         when (view?.id) {
             R.id.imgEdit -> {
-                //TODO intent to search
+                startActivity(Intent(activity, SearchLocationActivity::class.java))
             }
             R.id.btnConfirm -> {
```
Comment (at-quocnguyen) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: break line chia cho dễ nhìn cũng như đẹp không thì làm dòng comment mục đích làm j

**Comment by at-quocnguyen** [comment]
Update code mới nhất từ master  + cái này làm mừ chưa chạy đc đi dâu hả ??? test chưa a chỉ review code nên chưa chạy thử đc chưa nên hum sau làm mừ chờ j đó thì để dòng todo trên đầu class để a biết đường chạy thử nha!

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 12]
```diff
@@ -1,74 +1,152 @@
...
 
+    override fun onResume() {
+        super.onResume()
+        mGoogleApiClient.connect()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        mGoogleApiClient.disconnect()
+        if(!mLocationAsyncTask.isCancelled) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 13]
```diff
@@ -1,74 +1,152 @@
...
+            mLocationAsyncTask.cancel(true)
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        if(!mLocationAsyncTask.isCancelled) {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 14]
```diff
@@ -82,49 +160,38 @@ class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
...
-            tvLocation.text = address.getAddressLine(0)
-            if (!address.subThoroughfare.isNullOrEmpty()) {
-                mDestinationName = address.subThoroughfare.plus(" ").plus(address.thoroughfare)
-            } else {
-                mDestinationName = address.thoroughfare
-            }
-        } else {
-            tvLocation.text = null
-        }
+        mMapFragment = childFragmentManager.findFragmentById(R.id.fragmentConfirmMap) as SupportMapFragment
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: ERROR: Cast operator throws an exception if the cast is not possible.

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 15]
```diff
@@ -82,49 +160,38 @@ class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
...
-        mGoogleMap?.addMarker(MarkerOptions()
+        mGoogleMap.addMarker(MarkerOptions()
                 .position(latLng!!)
                 .icon(BitmapDescriptorFactory.fromResource(R.drawable.ic_ht_expected_place_marker))
                 .title(mDestinationName)
                 .anchor(0.5f, 0.5f))
                 ?.showInfoWindow()
         imgPickLocation.visibility = View.INVISIBLE
-        mGoogleMap?.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 16f))
+        mGoogleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 14f))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 16]
```diff
@@ -82,49 +160,38 @@ class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
...
-            mGoogleMap?.moveCamera(CameraUpdateFactory.newLatLngZoom(currentLocation, 16f))
-            addPulseRing(currentLocation)
-        }
+        val currentLocation = LatLng(location.latitude, location.longitude)
+        mGoogleMap.clear()
+        mMarker = mGoogleMap.addMarker(MarkerOptions()
+                .position(currentLocation)
+                .icon(BitmapDescriptorFactory.fromResource(R.drawable.ic_current_point))
+                .title(getString(R.string.current_location))
+                .anchor(0.5f, 0.5f))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt [hunk 17]
```diff
@@ -82,49 +160,38 @@ class ConfirmLocationFragment : BaseFragment(), OnMapReadyCallback,
...
+        addPulseRing(currentLocation)
+        mGoogleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(currentLocation, 16f))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/ConfirmLocationFragment.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/confirm/LocationNameAsyncTask.kt [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+import kotlinx.android.synthetic.main.view_confirm_location_layout.*
+import java.util.*
+
+/**
+ * Class get location name
+ * Created by haingoq on 30/10/2017.
+ */
+class LocationNameAsyncTask(private val mFragment: ConfirmLocationFragment) : AsyncTask<LatLng, Void, String>() {
+
+    override fun doInBackground(vararg latLng: LatLng?): String {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/confirm/LocationNameAsyncTask.kt]: WARNING: Excessive nesting leads to hidden complexity. Prefer extracting code to make it easier to understand.

**Pull request merged**